app/src/main/java/com/example/marsphotos/network/MarsApiService.kt

```diff
@@ -17,18 +17,20 @@
 package com.example.marsphotos.network
 
 import com.example.marsphotos.model.MarsPhoto
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType
 import retrofit2.Retrofit
-import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.http.GET
 
 private const val BASE_URL =
     "https://android-kotlin-fun-mars-server.appspot.com"
 
 /**
- * Use the Retrofit builder to build a retrofit object using a Gson converter
+ * Use the Retrofit builder to build a retrofit object using a kotlinx.serialization converter
  */
 private val retrofit = Retrofit.Builder()
-    .addConverterFactory(GsonConverterFactory.create())
+    .addConverterFactory(Json.asConverterFactory(MediaType.get("application/json")))
```

Comment:
The call to `asConverterFactory` produces an error as it's marked with `@ExperimentalSerializationApi`. This is because it makes calls into underlying APIs that have the same opt-in requirement, and it propagates it.

Can be suppressed by adding `@OptIn(ExperimentalSerializationApi::class)` above the call.